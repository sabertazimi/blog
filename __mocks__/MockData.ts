const time = new Date(2022, 0, 1, 8, 0, 0).toLocaleString('zh-CN', {
  hour12: false,
});

const siteMetadata = {
  title: 'Title',
  author: 'Sabertaz',
  siteUrl: 'https://example.com',
  email: 'example@github.com',
  disqusUrl: 'https://example.com',
  landingTitles: ['A', 'B', 'C'],
  socialList: {
    github: 'sabertazimi',
    twitter: 'sabertazimi',
    facebook: 'sabertazimi',
    linkedin: 'sabertazimi',
    weibo: 'sabertazimi',
  },
  bookList: Array.from(Array(3).keys()).map(() => ({
    title: 'Title',
    author: 'Sabertaz',
    url: 'https://example.com',
    description: 'Description',
  })),
};

const basePosts = Array.from(Array(5).keys()).map(index => ({
  index,
  fields: { slug: `/${index + 1}BasicNotes/` },
  frontmatter: { title: `${index + 1} Basic Notes` },
  timeToRead: index + 1,
}));

const posts = basePosts.map((post, index) => ({
  ...post,
  fields: { slug: `/${index + 1}BasicNotes/`, gitTime: `${index + 1} time` },
  frontmatter: {
    title: `${index + 1} Basic Notes`,
    subtitle: 'Be a Stupid Learner',
    author: 'Sabertaz',
    date: '2018-08-08T00:00:00.000Z',
    tags: ['JavaScript', 'Frontend Development', 'Web Development'],
  },
  timeToRead: index + 1,
}));

const MockData = {
  time,
  siteMetadata,
  basePosts,
  posts,
};

export default MockData;
