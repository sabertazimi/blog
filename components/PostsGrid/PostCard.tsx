import cx from 'classnames'
import styles from './PostCard.module.css'
import Container from '@/components/Container'
import { H2 } from '@/components/Headings'
import { Calendar, Read } from '@/components/Icons'
import Link from '@/components/Link'
import { Slide } from '@/components/Motion'
import Skeleton from '@/components/Skeleton'
import { IconTag, LinkTag } from '@/components/Tags'
import type { PostMeta } from '@/types'

interface Props {
  post: PostMeta
}

function PostCard({
  post: { slug, title, createTime, readingTime, tags },
}: Props): JSX.Element {
  return (
    <Container className="card mb-16 mt-0 px-5 py-4">
      <Slide delay={0.2}>
        <LinkTag tag={tags ? tags[0] : 'Computer Science'} />
        <H2 className={styles.title}>{title}</H2>
        <IconTag
          tag={createTime ? new Date(createTime).toDateString() : 'Nowadays'}
          icon={<Calendar />}
        />
        <Container className="mt-3">
          <Skeleton
            paragraph={{ rows: Math.min(Math.floor(readingTime / 2), 10) }}
          />
          <Link
            href={`/post/${slug}`}
            className={cx(
              'flex-container float-right m-0 h-14 w-14 rounded-full border-primary',
              'bg-primary text-2xl text-light',
              'transition hover:bg-secondary hover:text-light focus:bg-secondary focus:text-light',
            )}
          >
            <Read />
          </Link>
        </Container>
      </Slide>
    </Container>
  )
}

export default PostCard
