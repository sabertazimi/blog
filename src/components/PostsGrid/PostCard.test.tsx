import MockData from '@MockData';
import React from 'react';
import { create } from 'react-test-renderer';
import PostCard from './PostCard';

describe('PostCard', () => {
  const mockBasePost = MockData.basePosts[0];
  const mockPost = MockData.posts[0];

  test('should render correctly (snapshot)', () => {
    const tree = create(<PostCard post={mockPost} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('should render correctly with partial data (snapshot)', () => {
    const tree = create(<PostCard post={mockBasePost} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
