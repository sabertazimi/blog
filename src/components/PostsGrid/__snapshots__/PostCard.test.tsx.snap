// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard should render correctly (snapshot) 1`] = `
<div
  className="container relative block h-full mx-auto my-0 p-auto px-5 py-4 mt-0 mb-16 transition duration-300 shadow-xl transform-gpu hover:shadow-2xl hover:-translate-y-2"
  style={Object {}}
>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": " translateX(-200px)",
      }
    }
  >
    <span
      className="ant-tag ant-tag-has-color"
      style={
        Object {
          "backgroundColor": "#1890ff",
        }
      }
    >
      <a
        className="text-base font-extrabold"
        href="/tags/Redux"
      >
        Redux
      </a>
    </span>
    <h2
      className="ant-typography my-3"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Redux Basic Notes
    </h2>
    <span
      className="ant-tag tag-black"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <div
        className="text-base font-extrabold"
      >
        Posted on 
        Wed Aug 08 2018
         
      </div>
    </span>
    <div
      className="container relative block h-full mx-auto my-0 p-auto mt-3"
      style={Object {}}
    >
      <div
        className="ant-skeleton"
      >
        <div
          className="ant-skeleton-content"
        >
          <h3
            className="ant-skeleton-title"
            style={
              Object {
                "width": "38%",
              }
            }
          />
          <ul
            className="ant-skeleton-paragraph"
          >
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": "61%",
                }
              }
            />
          </ul>
        </div>
      </div>
      <button
        className="ant-btn ant-btn-default float-right m-0 text-2xl rounded-full w-14 h-14 button-primary"
        onClick={[Function]}
        type="button"
      >
        <a
          href="/reduxBasicNotes/"
        >
          <span
            aria-label="read"
            className="anticon anticon-read"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="read"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M928 161H699.2c-49.1 0-97.1 14.1-138.4 40.7L512 233l-48.8-31.3A255.2 255.2 0 00324.8 161H96c-17.7 0-32 14.3-32 32v568c0 17.7 14.3 32 32 32h228.8c49.1 0 97.1 14.1 138.4 40.7l44.4 28.6c1.3.8 2.8 1.3 4.3 1.3s3-.4 4.3-1.3l44.4-28.6C602 807.1 650.1 793 699.2 793H928c17.7 0 32-14.3 32-32V193c0-17.7-14.3-32-32-32zM324.8 721H136V233h188.8c35.4 0 69.8 10.1 99.5 29.2l48.8 31.3 6.9 4.5v462c-47.6-25.6-100.8-39-155.2-39zm563.2 0H699.2c-54.4 0-107.6 13.4-155.2 39V298l6.9-4.5 48.8-31.3c29.7-19.1 64.1-29.2 99.5-29.2H888v488zM396.9 361H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm223.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c0-4.1-3.2-7.5-7.1-7.5H627.1c-3.9 0-7.1 3.4-7.1 7.5zM396.9 501H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm416 0H627.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5z"
              />
            </svg>
          </span>
        </a>
      </button>
    </div>
  </div>
</div>
`;

exports[`PostCard should render correctly with partial data (snapshot) 1`] = `
<div
  className="container relative block h-full mx-auto my-0 p-auto px-5 py-4 mt-0 mb-16 transition duration-300 shadow-xl transform-gpu hover:shadow-2xl hover:-translate-y-2"
  style={Object {}}
>
  <div
    style={
      Object {
        "opacity": 0,
        "transform": " translateX(-200px)",
      }
    }
  >
    <span
      className="ant-tag ant-tag-has-color"
      style={
        Object {
          "backgroundColor": "#4263eb",
        }
      }
    >
      <a
        className="text-base font-extrabold"
        href="/tags/Computer Science"
      >
        Computer Science
      </a>
    </span>
    <h2
      className="ant-typography my-3"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      Redux Basic Notes
    </h2>
    <span
      className="ant-tag tag-black"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      <div
        className="text-base font-extrabold"
      >
        Posted on 
        Nowadays
         
      </div>
    </span>
    <div
      className="container relative block h-full mx-auto my-0 p-auto mt-3"
      style={Object {}}
    >
      <div
        className="ant-skeleton"
      >
        <div
          className="ant-skeleton-content"
        >
          <h3
            className="ant-skeleton-title"
            style={
              Object {
                "width": "38%",
              }
            }
          />
          <ul
            className="ant-skeleton-paragraph"
          >
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": undefined,
                }
              }
            />
            <li
              style={
                Object {
                  "width": "61%",
                }
              }
            />
          </ul>
        </div>
      </div>
      <button
        className="ant-btn ant-btn-default float-right m-0 text-2xl rounded-full w-14 h-14 button-primary"
        onClick={[Function]}
        type="button"
      >
        <a
          href="/reduxBasicNotes/"
        >
          <span
            aria-label="read"
            className="anticon anticon-read"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="read"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M928 161H699.2c-49.1 0-97.1 14.1-138.4 40.7L512 233l-48.8-31.3A255.2 255.2 0 00324.8 161H96c-17.7 0-32 14.3-32 32v568c0 17.7 14.3 32 32 32h228.8c49.1 0 97.1 14.1 138.4 40.7l44.4 28.6c1.3.8 2.8 1.3 4.3 1.3s3-.4 4.3-1.3l44.4-28.6C602 807.1 650.1 793 699.2 793H928c17.7 0 32-14.3 32-32V193c0-17.7-14.3-32-32-32zM324.8 721H136V233h188.8c35.4 0 69.8 10.1 99.5 29.2l48.8 31.3 6.9 4.5v462c-47.6-25.6-100.8-39-155.2-39zm563.2 0H699.2c-54.4 0-107.6 13.4-155.2 39V298l6.9-4.5 48.8-31.3c29.7-19.1 64.1-29.2 99.5-29.2H888v488zM396.9 361H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm223.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c0-4.1-3.2-7.5-7.1-7.5H627.1c-3.9 0-7.1 3.4-7.1 7.5zM396.9 501H211.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5zm416 0H627.1c-3.9 0-7.1 3.4-7.1 7.5v45c0 4.1 3.2 7.5 7.1 7.5h185.7c3.9 0 7.1-3.4 7.1-7.5v-45c.1-4.1-3.1-7.5-7-7.5z"
              />
            </svg>
          </span>
        </a>
      </button>
    </div>
  </div>
</div>
`;
